File: python/examples/orderbook_tickers.py
"""Batch orderbook tickers — best_bid / best_ask / midpoint per orderbook."""

import asyncio
import sys

from common import rest_client


async def main():
    client = rest_client()
    deposit_asset = sys.argv[1] if len(sys.argv) > 1 else None

    response = await client.metrics().orderbook_tickers(deposit_asset)

    print(f"orderbooks with tickers: {len(response.tickers)}")
    for entry in response.tickers[:10]:
        mid = entry.midpoint or "—"
        print(
            f"  {entry.orderbook_id} (market {entry.market_pubkey}, "
            f"outcome {entry.outcome_index}) mid={mid}"
        )

    await client.close()


asyncio.run(main())

File: typescript/examples/orderbook_tickers.ts
import { restClient, runExample } from "./common";

async function main() {
  const client = restClient();
  const depositAsset = process.argv[2];

  const response = await client.metrics().orderbookTickers(depositAsset);

  console.log(`orderbooks with tickers: ${response.tickers.length}`);
  for (const entry of response.tickers.slice(0, 10)) {
    const mid = entry.midpoint ?? "—";
    console.log(
      `  ${entry.orderbook_id} (market ${entry.market_pubkey}, outcome ${entry.outcome_index}) mid=${mid}`,
    );
  }
}

void runExample(main);
